Extended the test description in MS test to use a more friendly name …

samholder opened the pull request:
…for scenario outlines. This now includes the value from the first column of the examples, if unique or the 'Variant 0' name if this is not unique. this addresses  #490.

Added some tests around the MsTestGeneratorProvider, for which there were none, but the other generator providers don't seem to have many either.

darrencauthon commented on Generator/UnitTestFeatureGenerator.cs:
```diff
@@ -394,7 +394,7 @@ private void GenerateScenarioOutlineExamplesAsIndividualMethods(ScenarioOutline
         private void GenerateScenarioOutlineExamplesAsRowTests(TestClassGenerationContext generationContext, ScenarioOutline scenarioOutline, CodeMemberMethod scenatioOutlineTestMethod)
         {
-            SetupTestMethod(generationContext, scenatioOutlineTestMethod, scenarioOutline, null, rowTest: true);
+            SetupTestMethod(generationContext, scenatioOutlineTestMethod, scenarioOutline, null, null, null, rowTest: true);
```
This is just a code review from my iPhone, so I can't fiddle with the code to vet this idea before continuing... And this isn't a huge deal, but just a "red flag" that went off in my head...

Adding two more arguments to this method might make this a good time to consider a refactor -- especially if null is getting passed to them regularly. I'm one of those weirdos that thinks that if you're adding arguments to a mature method, there might be a problem.

samholder replied:
yeah I wondered about this as well, but I think the problem is that at the moment when you are generating a test for a scenario outline it creates the test method (with a name) and then changes the name after the method has been created. this also seemed like a smell to me and it seemed like it was caused by the method to setup the test method (SetupTestMethod) not having sufficient information passed to it to name the method without creating duplicates. I considered passing the name we wanted in, and whilst this would have been ok, it would have made if difficult to also generate the friendly name. As the friendly name is also part of the method it seemed sensible to pass the parts required to the method to generate the test method. happy for suggestions to change it though.

darrencauthon commented on Generator/UnitTestFeatureGenerator.cs:
```diff
@@ -561,22 +566,23 @@ private void GenerateTestBody(TestClassGenerationContext generationContext, Scen
-            testMethod.Attributes = MemberAttributes.Public;
-            testMethod.Name = string.Format(TEST_NAME_FORMAT, scenario.Title.ToIdentifier());
+            testMethod.Attributes = MemberAttributes.Public;
+            testMethod.Name=GetTestMethodName(scenario, variantName, exampleSetIdentifier);
```
Minor spacing

darrencauthon commented on Generator/UnitTestFeatureGenerator.cs:
```diff
@@ -561,22 +566,23 @@ private void GenerateTestBody(TestClassGenerationContext generationContext, Scen
+            if (variantName != null)
+            {
+                friendlyTestName = string.Format("{0}: {1}", scenario.Title, variantName);
+            }
```
Curly brace inconsistency.

For the record, I'm not being picky, I try to vet my own PRs in this same way. :tongue:

gasparnagy commented:
@samholder sorry. the gherkin3 PR caused a conflict on this, and since this is your fork i cannot easily fix it. Could you fix it or push it to a SpecFlow repo branch? (also Daren's comments)

samholder commented:
I'm currently away on a trip but yeah I'll fix this (and other comments
from Darren) when I get back home next week
On 3 Dec 2015 1:26 p.m., "Gáspár Nagy" <notifications@github.com> wrote:

> @samholder <https://github.com/samholder> sorry. the gherkin3 PR caused a
> conflict on this, and since this is your fork i cannot easily fix it. Could
> you fix it or push it to a SpecFlow repo branch? (also Daren's comments)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/techtalk/SpecFlow/pull/523#issuecomment-161639090>.
>

darrencauthon commented:
Just to note:  My comments are just comments... no need to hold this up over them. :smile:

samholder commented:
I'd like to do the clean up at least. :-)
On 3 Dec 2015 8:00 p.m., "Darren Cauthon" <notifications@github.com> wrote:

> Just to note: My comments are just comments... no need to hold this up
> over them. [image: :smile:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/techtalk/SpecFlow/pull/523#issuecomment-161766409>.
>

samholder commented:
I rebased my changes on the current v2 branch and fixed the compilation issues. Also fixed the code review changes, didn't think they were picky :smiley:
